### PR TITLE
Use live region in form component

### PR DESCRIPTION
**Pull request by pan-kot**
### Description

Replace aria-live with live-region component in form.

### How has this been tested?

Added new unit test, manual testing with VO and NVDA.

### Documentation changes

[*Do the changes include any API documentation changes?*]
- [ ] _Yes, this change contains documentation changes._
- [x] _No._

### Related Links

[*Attach any related links/pull request for this change*]

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- [ ] _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](CONTRIBUTING.md#public-apis)._
- [ ] _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](CONTRIBUTING.md#browsers-support)._
- [ ] _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- [ ] _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- [ ] _Changes are covered with new/existing unit tests?_
- [ ] _Changes are covered with new/existing integration tests?_
</details>

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/form/__tests__/form.test.tsx [hunk 1]
```diff
@@ -4,54 +4,63 @@ import React from 'react';
...
-      wrapper = renderForm({ errorText: '' });
+      ({ wrapper } = renderForm({ errorText: '' }));
       expect(wrapper.findError()).toBeNull();
     });
 
+    it('form error includes alert and live region', () => {
+      const { container } = renderForm({ errorText: 'Some error' });
+      const wrapper = createWrapper(container);
+
+      expect(wrapper.findAlert()!.getElement()).toHaveTextContent('Some error');
```
Comment (just-boris) [inline src/form/__tests__/form.test.tsx:53]: Why is this not `formWrapper.findError()`?

Comment (pan-kot) [reply src/form/__tests__/form.test.tsx:53]: The `findError` finds the error text which is inside the alert, see: https://github.com/cloudscape-design/components/blob/main/src/form/internal.tsx#L33

That was done to not break customers after the error icon has been introduced.

The new test only verifies the presence of the alert and live region and the `findError` selector is used in the above two scenarious.

Comment (just-boris) [reply src/form/__tests__/form.test.tsx:53]: Why not testing it like this

```js
const wrapper = renderForm({ errorText: 'Some error' });

expect(wrapper.findError()!.getElement()).toHaveTextContent('Some error');
// findByClassName works even on `FormWrapper`
expect(wrapper.findByClassName(liveRegionStyles.root)!.getElement()).toHaveTextContent('Error icon, Some error');
```

Comment (pan-kot) [reply src/form/__tests__/form.test.tsx:53]: Oh, I didn't consider that -- thanks!

**Review by just-boris — APPROVED** [review]

**Pull request merged**